pyswmm/links.py

```diff
@@ -0,0 +1,693 @@
+        ...     print link.qlimi
+        ...     link.qlimi = 0.2
+        ...     print link.qlimi
+        >>> 0
+        >>> 0.2
+        
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.qLimit, param)
+    
+    @property
+    def clossinlet(self):
+        """
+        Get Inlet Head Loss
+
+        :return: Paramater Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.q0
+        >>> 0
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.cLossInlet)
+
+    @clossinlet.setter
+    def clossinlet(self, param):
+        """
+        Set Link Inlet Head Loss
+
+        :param float param: New Parameter value
+  
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.cLossInlet
+        ...     link.cLossInlet = 0.2
+        ...     print link.cLossInlet
+        >>> 0
+        >>> 0.2        
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.cLossInlet, param)
+    
+    @property
+    def clossoutlet(self):
+        """
+        Get Outlet Head Loss
+
+        :return: Paramater Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.clossoutlet
+        >>> 0
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.cLossOutlet)
+    
+    @clossoutlet.setter
+    def clossoutlet(self, param):
+        """
+        Set Link Outlet Head Loss
+
+        :param float param: New Parameter value
+  
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.clossoutlet
+        ...     link.clossoutlet = 0.2
+        ...     print link.clossoutlet
+        >>> 0
+        >>> 0.2    
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.cLossOutlet, param)
+    
+    @property
+    def clossave(self):
+        """
+        Get Average Conduit Loss
+
+        :return: Paramater Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.clossave
+        >>> 0
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.cLossAvg)
+
+    @clossave.setter
+    def clossave(self, param):
```

Comment:
these names are hard to understand...

`average_head_loss` is already more clear than `clossave`